list/list_resource.go

```diff
@@ -41,6 +43,26 @@ type ListResource interface {
 	List(context.Context, ListRequest, *ListResultsStream)
 }
 
+// ListResourceWithRawV5Schemas is an interface type that extends ListResource to include a method
+// which allows provider developers to supply the ProtoV5 representations of resource and resource identity
+// schemas. This is necessary if list functionality is being used with a legacy resource.
```

Comment:
nit: Should we be more specific about what a legacy resource is? Or is that something we think provider developers will understand by reading this  

Perhaps just "a resource that is not defined in Framework"?🤔

Reply:
The latter suggestion is clearer 👍